app/models/form.rb

```diff
@@ -157,16 +157,25 @@ def depends_on_met(depends_on, case_log)
     depends_on.any? do |conditions_set|
       return false unless conditions_set
 
-      conditions_set.all? do |question, value|
+      conditions_set.all? do |condition|
+        object, method, value = condition.values_at("object", "method", "value")
+        object_instance = case object
+                          when "case_log"
+                            case_log
+                          when "user"
+                            nil
+                          end
+
         if value.is_a?(Hash) && value.key?("operator")
           operator = value["operator"]
           operand = value["operand"]
-          case_log[question]&.send(operator, operand)
+          object_instance[method]&.send(operator, operand)
```

Comment:
Could this blow up if object_instance is nil?

Reply:
Yes but that should never currently be the case (otherwise our existing code could also blow up)

Reply:
Fair enough, we could maybe add some logging here in the unlikely event it does blow up? Log the various values so we can easily debug if needed

Reply:
But it's not a strong opinion so happy to go with your call here